builtin/logical/transit/path_cache_config.go

```diff
@@ -0,0 +1,100 @@
+package transit
+
+import (
+	"context"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func (b *backend) pathCacheConfig() *framework.Path {
+	return &framework.Path{
+		Pattern: "cache-config",
+		Fields: map[string]*framework.FieldSchema{
+			"size": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Required:    false,
+				Default:     0,
+				Description: `Size of cache, use 0 for an unlimited cache size, defaults to 0`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathCacheConfigRead,
+				Summary:  "Returns the size of the active cache",
+			},
+
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathCacheConfigWrite,
+				Summary:  "Configures a new cache of the specified size",
+			},
+
+			logical.CreateOperation: &framework.PathOperation{
+				Callback: b.pathCacheConfigWrite,
+				Summary:  "Configures a new cache of the specified size",
+			},
+		},
+
+		HelpSynopsis:    pathCacheConfigHelpSyn,
+		HelpDescription: pathCacheConfigHelpDesc,
+	}
+}
+
+func (b *backend) pathCacheConfigWrite(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	// get target size
+	cacheSize := d.Get("size").(int)
+
+	// err if size is negative
+	if cacheSize < 0 {
+		return logical.ErrorResponse("size must be greater or equal to zero"), logical.ErrInvalidRequest
+	}
+
+	// convert the cache if the specified size is different from the size of the active cache
+	if cacheSize == b.lm.GetCacheSize() {
+		return nil, nil
+	}
+
+	if cacheSize == 0 {
+		b.lm.ConvertCacheToSyncmap()
+	} else {
+		err := b.lm.ConvertCacheToLRU(cacheSize)
+		if err != nil {
+			return nil, errwrap.Wrapf("failed to create cache: {{err}}", err)
+		}
+	}
```

Comment:
I wonder if lines 55->66 could be internal to the lock manager and only use a `lm.SetCacheSize(cacheSize)` function here?

Reply:
@briankassouf moved that logic [here](https://github.com/hashicorp/vault/pull/6225/files#diff-70530f22fc776fd9e11919754f7fa137R110)